From 4baaa3df79e545ddb3ce898423b81d1572c67fe1 Mon Sep 17 00:00:00 2001
From: rio004 <riyon336@gmail.com>
Date: Fri, 22 Dec 2023 12:09:20 +0000
Subject: [PATCH] mt6833-common: Correct dtb detection

Signed-off-by: rio004 <riyon336@gmail.com>
---
 BoardConfigCommon.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/BoardConfigCommon.mk b/BoardConfigCommon.mk
index 66212ec..a4f3611 100644
--- a/BoardConfigCommon.mk
+++ b/BoardConfigCommon.mk
@@ -47,7 +47,6 @@ BOARD_KERNEL_CMDLINE += androidboot.init_fatal_reboot_target=recovery
 BOARD_KERNEL_CMDLINE += androidboot.selinux=permissive
 
 BOARD_BOOTIMG_HEADER_VERSION := 2
-BOARD_INCLUDE_DTB_IN_BOOTIMG := true
 
 BOARD_KERNEL_BASE := 0x40078000
 BOARD_KERNEL_OFFSET := 0x00008000
@@ -60,6 +59,7 @@ BOARD_MKBOOTIMG_ARGS := --ramdisk_offset $(BOARD_RAMDISK_OFFSET)
 BOARD_MKBOOTIMG_ARGS += --tags_offset $(BOARD_KERNEL_TAGS_OFFSET)
 BOARD_MKBOOTIMG_ARGS += --dtb_offset $(BOARD_DTB_OFFSET)
 BOARD_MKBOOTIMG_ARGS += --header_version $(BOARD_BOOTIMG_HEADER_VERSION)
+BOARD_MKBOOTIMG_ARGS += --dtb $(TARGET_PREBUILT_DTB)
 
 # Bootloader
 TARGET_BOOTLOADER_BOARD_NAME := mt6833
@@ -78,7 +78,7 @@ TARGET_SCREEN_DENSITY := 440
 BOARD_HAVE_MTK_FM := true
 
 # Kernel
-BOARD_PREBUILT_DTBIMAGE_DIR := $(DEVICE_PATH)-kernel/dtb
+BOARD_PREBUILT_DTB := $(DEVICE_PATH)-kernel/dtb.img
 BOARD_PREBUILT_DTBOIMAGE := $(DEVICE_PATH)-kernel/dtbo.img
 TARGET_PREBUILT_KERNEL := $(DEVICE_PATH)-kernel/Image.gz
 
-- 
2.34.1

